apps/browser/qml/pages/components/Overlay.qml

```diff
@@ -500,64 +532,69 @@ Shared.Background {
                 }
             }
 
+            BookmarkModel {
+                id: bookmarkModel
+                activeUrl: toolBar.url
+            }
+
             Browser.HistoryList {
                 id: historyList
 
                 property int panelSize: favoriteGrid.contextMenu && favoriteGrid.contextMenu.active ? 0 : virtualKeyboardObserver.panelSize
 
                 width: parent.width
                 height: browserPage.height - _overlayHeight - panelSize
+                model: historyContainer.showHistoryList ? historyModel : 0
+                contentY: favoriteGrid.y
+                showDeleteButton: true
+
+                header: Browser.FavoriteGrid {
+                    id: favoriteGrid
+                    height: historyList.height
+                    opacity: visible && toolBar.opacity < 0.9 ? 1.0 : 0.0
+                    enabled: overlayAnimator.atTop
+                    visible: historyContainer.showFavorites
+                    _quickScrollRightMargin: -(browserPage.width - width) / 2
+
+                    header: Item {
+                        width: parent.width
+                        height: searchField.height + historyButton.height
+                    }
+
+                    model: BookmarkFilterModel {
+                        id: bookmarkFilterModel
+                        sourceModel: bookmarkModel
+                        maxDisplayedItems: search ? favoriteGrid.columns : bookmarkModel.count
```

Comment:
This is restricting search to a single row of Favorite items; is that correct?

Reply:
@llewelld  yes, this is restricting search to a single row of Favorite items